CodingPractice/PythonAssignments/cleancode/EmailExtract.py

```diff
@@ -0,0 +1,30 @@
+import re
+
+
+class MalformedEmailAddressException(Exception):
+    pass
+
+
+class EmailDetails:
+    def __init__(self, _id, _domain):
+        self.id = _id
+        self.domain = _domain
+
+    def __eq__(self, other):
+        return self.id == other.id and self.domain == other.domain
+
+    def get_id(self):
+        return self.id
+
+    def get_domain(self):
+        return self.domain
+
+
+def extract_email_address_details(email_addr):
+    try:
+        if not re.match(r'[^@]+@[^@]+\.[^@]+', email_addr):
+            raise MalformedEmailAddressException('This is an invalid email address')
```

Comment:
How about  
```raise MalformedEmailAddressException( email_addr )```

Reply:
Also it looks like you catch the Malformed exception raised at line 26 and then raise another one.

I think you can just do this to reraise the exception 
```python
except MalformedEmailAddressException:
    raise
except Exception:
    raise MalformedEmailAddressException(email_addr)
```